Jellyfin.Api/Controllers/NotificationsController.cs

```diff
@@ -0,0 +1,146 @@
+        /// </summary>
+        /// <param name="userId">The user's ID.</param>
+        /// <returns>Notifications summary for the user.</returns>
+        [HttpGet("{UserID}/Summary")]
+        [ProducesResponseType(typeof(NotificationsSummaryDto), StatusCodes.Status200OK)]
+        public IActionResult GetNotificationsSummary(
+            [FromRoute] string userId)
+        {
+            return Ok(new NotificationsSummaryDto());
+        }
+
+        /// <summary>
+        /// Endpoint for getting notification types.
+        /// </summary>
+        /// <returns>All notification types.</returns>
+        [HttpGet("Types")]
+        [ProducesResponseType(typeof(IEnumerable<NameIdPair>), StatusCodes.Status200OK)]
+        public IActionResult GetNotificationTypes()
+        {
+            return Ok(_notificationManager.GetNotificationTypes());
+        }
+
+        /// <summary>
+        /// Endpoint for getting notification services.
+        /// </summary>
+        /// <returns>All notification services.</returns>
+        [HttpGet("Services")]
+        [ProducesResponseType(typeof(IEnumerable<NameIdPair>), StatusCodes.Status200OK)]
+        public IActionResult GetNotificationServices()
+        {
+            return Ok(_notificationManager.GetNotificationServices());
+        }
+
+        /// <summary>
+        /// Endpoint to send a notification to all admins.
+        /// </summary>
+        /// <param name="name">The name of the notification.</param>
+        /// <param name="description">The description of the notification.</param>
+        /// <param name="url">The URL of the notification.</param>
+        /// <param name="level">The level of the notification.</param>
+        /// <returns>Status.</returns>
+        [HttpPost("Admin")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public void CreateAdminNotification(
+            [FromQuery] string name,
+            [FromQuery] string description,
+            [FromQuery] string? url,
+            [FromQuery] NotificationLevel? level)
+        {
+            var notification = new NotificationRequest
+            {
+                Name = name,
+                Description = description,
+                Url = url,
+                Level = level ?? NotificationLevel.Normal,
+                UserIds = _userManager.Users.Where(i => i.Policy.IsAdministrator).Select(i => i.Id).ToArray(),
+                Date = DateTime.UtcNow,
+            };
+
+            _notificationManager.SendNotification(notification, CancellationToken.None);
+        }
+
+        /// <summary>
+        /// Endpoint to set notifications as read.
+        /// </summary>
+        /// <param name="userId">The userID.</param>
+        /// <param name="ids">A comma-separated list of the IDs of notifications which should be set as read.</param>
+        [HttpPost("{UserID}/Read")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
```

Comment:
Have you confirmed if explicitly marking this as returning a `200` response does anything? It seems like the type of thing that would be assumed already by convention. I wouldn't expect this attribute to be required anywhere that isn't returning `IActionResult`

Reply:
So at the moment, it won't achieve anything functionally, because anything returning a DTO or void will automatically be tagged with a 200 response by the [api doc generator](https://github.com/domaindrivendev/Swashbuckle.AspNetCore#list-operation-responses), but I think its best to leave it in. It won't affect performance, and improves documentation for someone just reading the code. Plus, in the case of most of these endpoints, they aren't fully implemented, and when/if they are, they'll require adding more annotations because 404's will be possible responses to the user ID that most of them take in. If you disagree, I'm happy to remove them though, since it isn't strictly necessary.

Reply:
@mark-monteiro As for the documentation of that functionality, it's very unclear unfortunately. I essentially had to just infer it from [this page](https://docs.microsoft.com/en-us/aspnet/core/mvc/controllers/routing?view=aspnetcore-3.1#set-up-conventional-route). Sorry I couldn't be more help on that front.

Reply:
Yeah that makes sense to me, feel free to leave them in. Strange they don't have better documentation around the validations, but thanks for the links!